src/pcre2_compile_class.c

```diff
@@ -2467,33 +2475,229 @@ applications.
 
 The pptr will be left pointing at the matching META_CLASS_END. */
 
-BOOL
-PRIV(compile_class_nested)(uint32_t options, uint32_t xoptions, BOOL negated,
-  uint32_t **pptr, PCRE2_UCHAR **pcode, eclass_op_info *pop_info,
-  int *errorcodeptr, compile_block *cb, PCRE2_SIZE *lengthptr)
+static BOOL
+compile_eclass_nested(eclass_context *context, BOOL negated,
+  uint32_t **pptr, PCRE2_UCHAR **pcode,
+  eclass_op_info *pop_info, PCRE2_SIZE *lengthptr)
 {
 uint32_t *ptr = *pptr;
-PCRE2_UCHAR *code = *pcode;
 #ifdef PCRE2_DEBUG
 PCRE2_UCHAR *start_code = *pcode;
 #endif
 
 /* The CLASS_IS_ECLASS bit must be set since it is a nested class. */
-PCRE2_ASSERT(*ptr == (META_CLASS | CLASS_IS_ECLASS) ||
-             *ptr == (META_CLASS_NOT | CLASS_IS_ECLASS));
+PCRE2_ASSERT(META_CODE(*ptr) == META_CLASS ||
+             META_CODE(*ptr) == META_CLASS_NOT);
 
-if (*ptr++ == (META_CLASS_NOT | CLASS_IS_ECLASS))
+if (META_CODE(*ptr) == META_CLASS_NOT)
   negated = !negated;
 
 /* Because it's a non-empty class, there must be an operand at the start. */
-if (!compile_class_binary_loose(options, xoptions, negated, &ptr, &code,
-                                pop_info, errorcodeptr, cb, lengthptr))
+(*pptr)++;
```

Comment:
This `(*pptr++)` is placed confusingly under the comment now. But this skip has nothing to do with the operand, it's the META_CLASS we're skipping.